src/gsy_e/models/strategy/energy_parameters/energy_params_eb.py

```diff
@@ -21,37 +21,190 @@
 NOTE: The intraday product should not use this approach, since it already trades on single
 15-minutes timeslots.
 """
-from typing import Optional
+from abc import ABC, abstractmethod
+from typing import Optional, TYPE_CHECKING, Dict
 
 import pendulum
 from gsy_framework.enums import AvailableMarketTypes
 from gsy_framework.forward_markets.forward_profile import ForwardTradeProfileGenerator
 from gsy_framework.utils import convert_kW_to_kWh
 
+from gsy_e.constants import FLOATING_POINT_TOLERANCE
 from gsy_e.models.state import LoadState, PVState
 
+if TYPE_CHECKING:
+    from gsy_e.models.state import StateInterface
 
-class ConsumptionStandardProfileEnergyParameters:
+
+class ForwardEnergyParams(ABC):
+    """Common abstract base class for the energy parameters of the forward strategies."""
+    def __init__(self):
+        self._posted_energy_kWh: Dict = {}
+
+    def get_posted_energy_kWh(
+            self, market_slot: pendulum.DateTime, market_type: AvailableMarketTypes) -> float:
+        """Retrieve the already posted energy on this market slot."""
+        if market_type == AvailableMarketTypes.INTRADAY:
```

Comment:
I thought about it originally too, however it looks like an overkill to my eyes, because there are only 2 methods that share this behavior (get_available_energy and event_traded_energy). If you insist and think that it is worth it let me know and I will give it a try.

Reply:
AFAIS the conditionals are also in `increment_posted_energy()`, `decrement_posted_energy()` and `get_posted_energy_kWh()` (in the parent class). In my opinion it would make sense to find a way to "simplify" these by moving the conditional logic into different classes. We can also wait for other opinions of course

Reply:
Indeed, I did not check the base class methods ! I have refactored it already then, please have a look and let me know what you think, thanks!